src/main/resources/assets/js/launcher/main.ts

```diff
@@ -235,6 +230,20 @@ class Launcher {
         return `theme-${ThemeManager.getTheme(this.config['theme'])}`;
     };
 
+    private getLauncherUrl(): string {
+        const pathName = window.location.pathname;
+        const adminBaseUrl = this.config['adminBaseUrl'];
+
+        return pathName === `${adminBaseUrl}` || pathName === `${adminBaseUrl}/`
+                  ? UriHelper.joinPath(adminBaseUrl, `/${homeAppToolKey}/home/_/${homeAppToolKey}/launcher`)
+                  : UriHelper.joinPath(pathName, `/_/${homeAppToolKey}/launcher`);
+    }
+
+    private isHomeApp(): boolean {
```

Comment:
what is so special about home app so it needs to be customized in almost every method?